### ISSUE TITLE
Brave Browser Soundcloud login not working

### ISSUE THREAD

**Issue by Crystallic** [brave/brave-browser#27117]
I know there are some similar issues but they do not describe this problem directly.

It's not possible to login on soundcloud.com with brave browser, even with disabled security settings. I do not use google, facebook, apple or any other external account login. I use a dedicated account created for soundcloud with my mailaddress.

![image](https://user-images.githubusercontent.com/35669845/205285386-e2ee41c4-c547-46ec-bee9-6e9769cc41c5.png)

Every time after pressing the "Sign In" button I get the the error message "Something unexpected happened. Please try again."
I already disabled all securtity plugins, disabled all brave internal security settings, cleared all cookies related to soundcloud and retstarted the browser multiple times. I also tried to login on different ways e.g. pop-up when clicking on sign in, by navigating manually to soundcloud.com/signin and so on.

Since I'm searching now for some days for any solution and can not find anything that helps on google, github etc., I hope asking you directly will deliver a working solution. Maybe it's something with the browser it self but then I'm wondering why no one else has ever posted anything about that. So I guess it's me how is doing something wrong since I'm very new to brave and evaluating the browser atm as alternative for firefox.

Maybe someone here knows what is wrong and can help me as well as other folks with that issue to solve it or to find a working solution/workarround.

Thanks a lot

**Comment by Gontinoc** [comment]
I have the exact same problem and I tried every step as the OC. 
Maybe it's noteworthy that this only occurs while using my Macbook Pro. If I try to log in on my android phone using brave browser, it works fine.
Also after I clear cache and cookies, I get the error that "their bot thinks I am a bot".
Then it's the error as mentioned in OC.

Some help would be awesome.

**Comment by Crystallic** [comment]
Yeah, I'm also on Mac & Linux. For Both the same but don't know what is on Windows, I don't use that.
What I can tell on mobile device it works w/o any issue. I have tried it an iPhone & Android und with both devices I can login directly.
Is it a bug for desktop environments in general or maybe is it related to the *nix system layout or architecture? 

Hope this will be solved soon. In the meantime brave won't fit for me as primary browser on a daily bases and will stick to firefox furthermore.

**Comment by ryanbr** [comment]
Still an issue? @Crystallic @Gontinoc 

Just tested Brave release & Nighly, using email/password logins. With default shields, login worked

**Comment by Crystallic** [comment]
What system you're on? Also on MacOS/Linux? Still not work.. Tried Beta, Brave & Nightly release with shilds and w/o them and a lot more different settings  for each of them Even on three different OS's (MacOSX Ventura 13.0.1, ArchLinux & Debian 11 Bullseye).. always the same  result. @ryanbr 
I'm very interested how you do it. Because I tried it now for hours and different days w/o any luck. I guess found what's going on and if I'm right there are just two possible solutions how you could get it to work again (as long the website didn't fixed it).
1. to downgrade your browser to a time bevor the release
2. to have a working solution arround that new implementitions from Google like bypassing it or least the still not fixed website(s)

But first of all one BIG COMPLIMENT to the Brave team (@brave-browser-releases @brave-builds @brave-dev)! You make for each release an independent application so that's possible to use them all at once w/o deleting or even closing the other version. I love you very much for that decision guys!

I found when I tried to debug a little w/ browser developer tools etc. something what looks like the issue here. I also verified that's not just a "Brave" issue it's related to Chromium and all Chromium based browsers by downloading some of them, in detail chromium it self (in two different versions) and Vivaldi. All versions have the same issue!
```
A page or script is accessing at least one of navigator.userAgent, navigator.appVersion, and navigator.platform. Starting in Chrome 101, the amount of information available in the User Agent string will be reduced.
To fix this issue, replace the usage of navigator.userAgent, navigator.appVersion, and navigator.platform with feature detection, progressive enhancement, or migrate to navigator.userAgentData.
Note that for performance reasons, only the first access to one of the properties is shown.
```
On hitting the "signin" button there is a javascript what's using nowadays forbidden methods and after a short time I found some threads what describe that Google implementet this "new security feature" (how they call it) what doesn't allow it anymore for Chromium (& obvoiusly Chromium based browser).. Mobil Devices will follow soon..

And if you search foexacly that terms you will find a lot more who had that issue in diffenrent sites. The only solution seems to mirgrate to the new path.

Here the reference the anouncements
https://blog.chromium.org/2021/09/user-agent-reduction-origin-trial-and-dates.html
https://blog.chromium.org/2021/05/update-on-user-agent-string-reduction.html
https://chromiumdash.appspot.com/schedule

Thanks a lot Google, didn't know I could be even more unsattisfied how this company disrespects users all over the world and their privacy and just try to prevent others from it now. I guess just to sell them the data in the end. 

Opened a report by soundclound with a fix request some details about what and where. Hopefully they will be fast to fix that (It's just re-label some object name with the new name as it looks to me)

I might be wrong since I'm not a developer just an Engineer in the departments of Linux, security, k8s, cloud & automation. So it's possible that I've just miss intrepreted something here, but It stinks very hard to be the issue here I think.

**Comment by ryanbr** [comment]
If I enabled "Aggressive fingerprinting" I'll get an error on login. If its standard shields/standard fingerprinting. Works here @Crystallic  Tested on MacOS (M1). 
![20230110_141519](https://user-images.githubusercontent.com/1659004/211439981-d391ab43-1553-4300-9743-6363b93473ea.jpg)

**Comment by Crystallic** [comment]
Yeah, tried that as well also on M1 Mac but for it nerver worked for me. I got just the Error from your screenshot but it won't log me in. Interesting you were more lucky then me. What's the device you choose for that spoofing? I tried just Apple device as iPad, iPhone etc. and the default preconfigured Android device (aleady forgot what it is) so I'm just currious what you did to get a different result them for days. Because tried for many days always when I was really anoyed about the missing features w/o account I tried a lots of stuff to get in. I tried even in addition to that print spoofing to combine it with spoofed user agent to get rid of that rebot detection what always prevented a login by acting as mobile device
And one more question..
> Still an issue? @Crystallic @Gontinoc
> 
> Just tested Brave release & Nighly, using email/password logins. With default shields, login worked

why saying it works if default settings and didn't mention what you really did? In facts it is something complete different as a regular login. Actually you kinda hacked yourself in instead of passing the default login (not w/ forbidden or illegal methodes but still). Why do not share that really important part of the information how "your login worked"?

I just hope Soundcloud will answer me asap and fix that issue. I sent them a detailed Information about what I found and how to fix (even with description of the codes location). That could be provided very fast if they pay attantion to it. Actually, would Soundcloud be "Open Source" it could be done already by now.. just some one has to create the PR  and someone else to git on "merge". I'm even truly convince they would not have such issues and improved security.
But I guess I'm dreaming again :( So let's hope thay have also the interests to get it done, then I belive it will be done in less then but they have to start first.

**Comment by ryanbr** [comment]
> why saying it works if default settings and didn't mention what you really did? I don't change anything of the settings.

Default shields:
```
Ads & Trackers = Standard
Upgraded https = enabled
Blocked Scripts = disabled
Standard fingerprinting
Block Cross site cookies
```

Would test soundcloud in private window mode. With the above shield settings.

**Comment by Gontinoc** [comment]
I still get the same error:
"Something unexpected happened. Please try again."


Am 11. Jan. 2023, 02:28 +0100 schrieb Fanboynz ***@***.***>:

**Comment by Crystallic** [comment]
Solved as we have working solutions to run Soundcloud on MacOSX & Linux desktop machines (the Open Source Community is again the life saver as most of the time (or should I say "as always" 😉 ) - they offer SoundCleod & Soundnode, what is just SoundCloud even with the same UI feeling etc. but delivered as Open Source frontend appliaction that serves 100% original Soundcloud data in their backend). Actually works the solution except of a little mess with some colors, what is not even disturbing anymore after a short time of ussage, much cleaner then the original SoundCloud App performance / resources wise. Also I don'f feel bad a scared to enter my credential since it is absolute Open Source (sure there are possiblities but both apps are available for a long time now and used very often what eleminates the chance of potential data theft etc to a minimum of risk).

**It's also not even a "Brave Browser" issue.**

It's actually an issue what Soundcloud has to solve, but it seems there is no prio at all for it by SoundCloud. They do not react to nothing, ignore every request & have not even one statement at all about that topic. Feels like they don't care about it.

**Comment by Gontinoc** [comment]
Thanks @Crystallic.
I can’t get soundnode to work, it gets stuck after launching, trying to authenticate my Soundcloud account. But soundcleod works.
The solution still forces me to leave brave browser and use the app though. So that’s basically the same as using another browser ;)
But still, thanks :)
Am 13. Jan. 2023, 18:07 +0100 schrieb Crystallic ***@***.***>:

**Comment by Crystallic** [comment]
> Thanks @Crystallic. I can’t get soundnode to work, it gets stuck after launching, trying to authenticate my Soundcloud account. But soundcleod works. The solution still forces me to leave brave browser and use the app though. So that’s basically the same as using another browser ;) But still, thanks :) Am 13. Jan. 2023, 18:07 +0100 schrieb Crystallic ***@***.***>:
> […](#)
> Solved as we have working solutions to run Soundcloud on MacOSX & Linux desktop machines (the Open Source Community is again the life saver as most of the time (or should I say "as always" 😉 ) - they offer SoundCleod & Soundnode, what is just SoundCloud even with the same UI feeling etc. but delivered as Open Source frontend appliaction that serves 100% original Soundcloud data in their backend). Actually works the solution except of a little mess with some colors, what is not even disturbing anymore after a short time of ussage, much cleaner then the original SoundCloud App performance / resources wise. Also I don'f feel bad a scared to enter my credential since it is absolute Open Source (sure there are possiblities but both apps are available for a long time now and used very often what eleminates the chance of potential data theft etc to a minimum of risk). It's also not even a "Brave Browser" issue. It's actually an issue what Soundcloud has to solve, but it seems there is no prio at all for it by SoundCloud. They do not react to nothing, ignore every request & have not even one statement at all about that topic. Feels like they don't care about it. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: ***@***.***>

Hey @Gontinoc ,
I guess you're on M1 MacBook? Then you skipped the important part where I mentioned exactly what you are talking about right now. For me it's the same with m1 and Soundnode therefor I have given a step by step explaination how to install SoundCleod from commandline using homebrew.
 

TLDR;
For everyone who has just interessts in the actuall solution to get at running and don't care about the theoretical background & historical lesson. 
At the bottom are the two commands you have to execute. That's all you have to do!



If you don't know what homebrew is, let me explain it with a few words so that you get an idea about the purpose of that tool. 
Apples operating system the OSX is in it's base as Linux, actually they have the same ancestors and are both members of the to *NIX family. In the past (not so long ago) Linux was for the most folks not a usable system im the daily life but there was Apple with the propertary OSX whats kinda the same but although not. OSX was (you could say) always a reliable operating system with the ability to serve the avery user in a daily life. It just had and still has one very big disadvantage; It's closed up very hard (in the past even harder, they had also their completely own developed interfaces) was is e.g. the reason you need very often some specific hardware for OSX devices because the deafult what runs on every other system available was not compatible with Apples devices though.. (they define the income over the hardware instead of software as e.g. Microsoft) 

With linux on the other hand you can just do what you want to do but you have to make it all by yourself.. nowaday with the rasing times of Open Source and also Linux systems (even microsoft developed already two linux distributions and ther their windows couldn't be more different in all aspects or in general all their products from the past) 

so what has this all todo with homebrew? 
In Linux you need any tool the first thing you do is asking the package manager if the is something usefull in one of the defines repositories and if it's not or ist is just a very old release available that's not working for your case, the next step is very often just to add the software maintainers repository to your package manager. After that the offiziel and recent releases are directly available in you OS over the integrated package manager (any linux "family" has it's own btw.. the is debian with all of it's sub distros starting with ubuntu and a lot more all using the "apt" or "apt-get", Red Hat system like fedory, CentOS, Rocky, Alma etc used till RHEL major release 7 "yum" and changed to a new developed one called "dnf" since major release 8.. I could mention even a lot more and starting it's own topic about the diffent package manager some system have very spezialist implementations but anyways)  even with a lot of different implementations they have all one in common, you  have just to provide the right sources (could be a repository, a source package itself, a tarball etc) and by typing a few commands in your terminal you get get almost everything for your system. 
homebew os just like those "linux packagemanger" but it also works on Apples OSX. This empowers you to be able to install a lot of *nix compatible packages w/o the need of the appstore or any website you have to search for serving a feasable install package, just by typing 3 words in your terminal like linux users do it all day. E.g you need the simple commandline tool to make a web request to download a something from the commandline called "wget". Just typ "brew install wget" in the terminal and wait until all is setup for your.

How to prepare the system to make homebrew working with it?
actually it's just one line of code you have to execute in the terminal. (for more informations about that rach out tp their website https://homebrew.sh)
the command you have to excute would be
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`

after it's finshed you are already able to use homebrew. 
to get SoundCleod it's just one more step and you are ready to use it.

just execute (on your commandline again) the following comand:
`brew install soundcleod`

when it's done you can find it by all your apps over the launcher.


I hope could help you to understand background of what we do here and why. I know it's a lot of theory for just two command executions but after explaining it w/o the context (a few posts before) perple get maybe confused about the need of the terminal and just skip it.

If you have (or someone else) still questions about it or the need of some help would be necessary to get it done don't hesitate to ask again. there are no dumb questions, just dumb answers! so ask again and again till you to understand everything and there is no question left.


Here again at once all steps to do to get Homebrew and SoundCleod running on your system
```

# Installation of Hombrew
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

# Installation SoundCleod via Homebrew
brew install soundcleod
``` 


(I'm very sry about all typos, gramitical issues and so but It's 2 a.m. here (in germany) by now and I'm to tired clean thus all up)

@Gontinoc if you still need some support with it I'll offer you to help you with getting it done in a life session over Discord, Teams, "Brave Talk" or what you would like to use. The timing is just not the best since I am going to party this evening and have to sleep abviously today, too. So there will be a small window for it today otherwise I'm available on Monday again.

**Comment by Crystallic** [comment]
Alternatively you can use any browser that's not depends Chromium! My Proposal would be Mozilla Firefox. 
But I guess most folks don't want to run 2 or more web browser at once and like me you already noticed that enormous speed advantage in brave against other browsers as well as the integrated Adblocker etc. you would not like to give it up again.

Then one of the solutions above should serve you well. (M1 Mac has to rely on the terminal installation of SoundCleod since Soundnode it loading for ever at start)

**Comment by Gontinoc** [comment]
Wow. Thanks for the detailed explanation.
I'll be honest. I don't understand much of this tech stuff.
For me there was no need for a different kind of installation. I simply downloaded the soundcleod dmg and done. Worked like a charm.

And btw, I am from Germany as well :)
Am 14. Jan. 2023, 02:36 +0100 schrieb Crystallic ***@***.***>:

**Comment by Crystallic** [comment]
I'm gald to hear it's working for you as well by now. I'm not sure anymore what's the reason I did the install over cli but there was some 😄 Anyways, it works and we can enjoy listening to soundcloud again 🎵

**Comment by AgrippinaBausch** [comment]
Hi all, I had the same problem after formatting my laptop with soundcloud and some other websites. Either it wouldn't recognize my email/account name or it would only allow me the "sign-up" option.

The way it worked for me was to go to Password Manager and manually add the username and password for the website I wanted.
